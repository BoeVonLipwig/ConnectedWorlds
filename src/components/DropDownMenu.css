.menu {
  z-index: 9;
  display: none;
}

.menu-selected {
  outline: var(--fg) dotted thin;
}

.drop-down {
  float: left;
  margin-top: ;
}

#menu {
  position: absolute;
  padding-left: 0;
  z-index: 9;
  color: var(--fg);
  background-color: var(--bg);
  border: solid 1px var(--ll);
  font: 15px roboto condensed, helvetica neue, helvetica, arial, sans-serif;
  list-style: none;
}

.menu-open {
  display: block;
}

.menu::-webkit-scrollbar-track {
  background-color: var(--bg);
}

.menu::-webkit-scrollbar {
  width: 12px;
  background-color: var(--fg);
}

.menu::-webkit-scrollbar-thumb {
  background-color: var(--fg);
}

.ui-menu-item-wrapper {
  padding: 12px 20px;
}

.ui-menu-item:hover {
  background-color: var(--fg);
  color: var(--bg);
}

.ui-menu-item:hover .svg-path {
  filter: invert(100%);
}

.ui-menu-item .svg-path {
  fill: var(--fg);
}
