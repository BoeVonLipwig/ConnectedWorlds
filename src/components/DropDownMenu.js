import React, { Fragment } from "react";
import "./DropDownMenu.css";
import MenuButton from "./MenuButton";
import cytoscapeStore from "../util/CytoscapeStore";

class DropDownMenu extends React.Component {
  constructor(props) {
    super(props);
    if (this.props.clickHandler) {
      this.buttonClicked = this.props.clickHandler;
    }
    this.setWrapperRef = this.setWrapperRef.bind(this);
    this.handleClickOutside = this.handleClickOutside.bind(this);
    this.state = {
      showMenu: false
    };
  }

  onItemSelected = name => {
    cytoscapeStore.focusType = name.toLowerCase();
    this.setState({ ...this.state, showMenu: false });
  };

  render() {
    let divClass = "";
    if (this.state.showMenu) {
      divClass = "menu-selected";
    }
    return (
      <div ref={this.setWrapperRef} className="control">
        <div className={divClass}>
          <MenuButton name="Menu" onSelect={this.buttonClicked} />
        </div>
        {this.state.showMenu ? this.createMenu() : null}
      </div>
    );
  }

  createMenu = () => {
    return (
      <div className="menu menu-open menu-div">
        <ul id="menu" className="menu-ul">
          {this.allButtons()}
        </ul>
      </div>
    );
  };

  allButtons() {
    return (
      <Fragment>
        {this.props.data.map(buttonName => (
          <MenuButton name={buttonName} onSelect={this.onItemSelected} />
        ))}
      </Fragment>
    );
  }

  componentDidMount() {
    document.addEventListener("mousedown", this.handleClickOutside);
  }

  // noinspection SpellCheckingInspection
  componentWillUnmount() {
    document.removeEventListener("mousedown", this.handleClickOutside);
  }

  /**
   * Set the wrapper ref
   */
  setWrapperRef(node) {
    this.wrapperRef = node;
  }

  /**
   * Alert if clicked on outside of element
   */
  handleClickOutside(event) {
    if (this.wrapperRef && !this.wrapperRef.contains(event.target)) {
      this.setState({ ...this.state, showMenu: false });
    }
  }

  buttonClicked = () => {
    this.setState({ ...this.state, showMenu: !this.state.showMenu });
  };
}

export default DropDownMenu;
