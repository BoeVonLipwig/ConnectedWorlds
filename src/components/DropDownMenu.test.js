import React from "react";
import ReactDOM from "react-dom";
import DropDownMenu from "./DropDownMenu";
import { shallow } from "enzyme";

it("renders button without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<DropDownMenu />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders button correctly without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<DropDownMenu />, div);
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});
