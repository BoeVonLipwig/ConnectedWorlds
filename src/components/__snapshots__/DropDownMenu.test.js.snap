// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button correctly without crashing 1`] = `
<div>
  <div
    class="control"
  >
    <div
      id="drop-down-menu"
      class=""
    >
      <div
        class="button-css"
      />
    </div>
  </div>
</div>
`;
